File: model/src/main/java/io/github/runtimemodels/aop/validation/ValidationModule.java
package io.github.runtimemodels.aop.validation;

import com.google.inject.AbstractModule;

import static com.google.inject.matcher.Matchers.annotatedWith;
import static com.google.inject.matcher.Matchers.not;

/**
 * The {@linkplain ValidationModule} class provides a Guice binding module for validations.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public class ValidationModule extends AbstractModule {

    @Override
    protected void configure() {
        bindInterceptor(not(annotatedWith(DoNotCheck.class)), not(annotatedWith(DoNotCheck.class)), new CheckNotNull());
    }

}

File: chazm-model/src/main/java/runtimemodels/chazm/model/aop/validation/ValidationModule.kt
package runtimemodels.chazm.model.aop.validation

import com.google.inject.AbstractModule

import com.google.inject.matcher.Matchers.annotatedWith
import com.google.inject.matcher.Matchers.not
import java.lang.reflect.Method

/**
 * The [ValidationModule] class provides a Guice binding module for validations.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
class ValidationModule : AbstractModule() {

    override fun configure() {
        bindInterceptor(not(annotatedWith(DoNotCheck::class.java)), not<Method>(annotatedWith(DoNotCheck::class.java)), CheckNotNull())
    }

}
